Sources/UserDefaultsGeneratorCore/ConfigurationParser.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+import Yams
+
+public enum SwiftType: String, Decodable, Comparable {
+    public static func < (lhs: SwiftType, rhs: SwiftType) -> Bool {
+        return lhs.rawValue < rhs.rawValue
+    }
+    
+    case Int
+    case Bool
+    
+    var getterMethodName: String {
+        switch self {
+        case .Int:
+            return "integer"
+        case .Bool:
+            return "bool"
+        }
+    }
+}
+
+public struct Configuration: Decodable {
+    public let name: String
+    public let type: SwiftType
+    public let key: String?
+}
+
+public protocol ConfigurationParser {
+    func parse() throws -> [Configuration]
+}
+
+public struct YamlParser: ConfigurationParser {
```

Comment:
In Swift way, you should name `YAMLParser`

Reply:
:tashikani: :yosasou:

Reply:
:done: